### PR TITLE
Make Searching functional based on city

### LINKED ISSUE #48: FS: Dynamic dashboard
Make the dashboard dynamic and make searching functional based on city

### PR DESCRIPTION
**Pull request by RetiFier**
### What this PR does (required):
- Create Pet Sitter Controller 
- Make Searching functional based on city

### Screenshots / Videos (front-end only):
image
image

### Any information needed to test this feature (required):
- http://localhost:3001/petsitter for create petsitters and get petsitters
- http://localhost:3001/petsitter?address=city to search petsitter by city
- http://localhost:3001/petsitter/:id/edit to edit petsitters data such as rate, availability
- Should display search results in the user interface

### Any issues with the current functionality (optional):
- The availability search is not yet complete due to the development of the availability controller.
- The schedule base search is already implemented in the backend, but it needs to be fixed depending on availability.

### CODE REVIEW

File: server/controllers/petSitter.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+const asyncHandler = require("express-async-handler");
+const PetSitter = require("../models/PetSitter");
+const { Schedule } = require("../models/Availability");
+
+// @route POST /petsitter
+// @desc create a new petsitter
+// @access private
```
Comment (sundayezeilo) [inline server/controllers/petSitter.js:7]: Nice documentation

File: server/controllers/petSitter.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    rate,
+    description,
+    photo,
+  });
+  return res.status(201).json({ petsitter: newPetSitter });
+});
+
+// @route GET /petsitter
+// @desc Get All petsitter and Search petsitter
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/petSitter.js:42]: :+1:

File: server/controllers/petSitter.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    res.status(404);
+    throw new Error("Pet Sitter Not Found");
+  } else {
+    petSitter.set({
+      stripeConnectId,
+      availabilityId,
+      activatedAvailabilitySchedule,
+      rate,
+    });
+    const updatedPetSitter = await petSitter.save();
```
Comment (sundayezeilo) [inline server/controllers/petSitter.js]: This approach would set any specified field that did not exist in the request body to `undefined`. You can simply use mongoose `findOneAndUpdate` to update your document.

File: server/routes/petSitter.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const express = require("express");
+const router = express.Router();
+const { protect } = require("../middleware/auth");
+const { createPetSitter, getAllPetSitter, updatePetSitterById } = require("../controllers/petSitter");
+
+router.route("/").get(protect, getAllPetSitter).post(protect, createPetSitter);
+router.route("/:id/edit").put(protect, updatePetSitterById);
```
Comment (sundayezeilo) [inline server/routes/petSitter.js]: Use a `PATCH` request here if you're not updating the whole resource

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Good progress on this one, @RetiFier. I left you a few comments to address.
Happy coding!

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please address merge conflicts

**Review by moffatethan — APPROVED** [review]

**Pull request merged**